automation/system_test/prepare.py

```diff
@@ -372,11 +424,25 @@ def _patch_mlrun(self, provctl_path):
                 "create-patch",
                 "appservice",
                 override_image_arg,
+                "--gzip-flag=-1",
+                "-v",
                 f"--target-iguazio-version={str(self._iguazio_version)}",
                 "mlrun",
                 self._mlrun_version,
                 mlrun_archive,
             ],
+            detach=True,
+        )
+        self._run_and_wait_until_successful(
+            command=f"if cat {str(self.Constants.workdir)}/provctl-create-patch-{time_string}.log | "
+            f'grep "Patch archive prepared"; then echo "True"; else exit 1;fi',
```

Comment:
`grep` already exits with an error code if it doesn't find what you are looking for, so you can change this to:
```suggestion
            command=f"grep 'Patch archive prepared' {str(self.Constants.workdir)}/provctl-create-patch-{time_string}.log",
```